grudge/trace_pair.py

```diff
@@ -419,7 +419,8 @@ def communicate_single_array(key, local_bdry_ary):
                     local_bdry_ary, dest_rank=remote_rank, comm_tag=ary_tag,
                     stapled_to=make_distributed_recv(
                         src_rank=remote_rank, comm_tag=ary_tag,
-                        shape=local_bdry_ary.shape, dtype=local_bdry_ary.dtype))
+                        shape=local_bdry_ary.shape, dtype=local_bdry_ary.dtype,
+                        axes=local_bdry_ary.axes))
```

Comment:
Needs https://github.com/inducer/pytato/pull/339

Reply:
Thanks! That's in now, so let's see how we look.